python_modules/dagster/dagster_tests/storage_tests/utils/event_log_storage.py

```diff
@@ -5904,3 +5908,287 @@ def test_asset_tags_to_insert(self, test_run_id: str, storage: EventLogStorage):
+            ) == {"3"}
+            assert (
+                storage.get_updated_data_version_partitions(
+                    asset_key, partitions=partitions, since_storage_id=after_three
+                )
+                == set()
+            )
+
+    def test_updated_none_data_version(self, storage, instance):
+        asset_key = AssetKey(["one"])
+        partitions = ["1", "2", "3"]
+
+        def _materialize_partition(partition, data_version):
+            tags = {"dagster/data_version": data_version} if data_version else None
+            yield AssetMaterialization(asset_key=asset_key, partition=partition, tags=tags)
+
+        def _materialize(data_version):
+            for partition in partitions:
+                yield from _materialize_partition(partition, data_version)
+
+        @op
+        def materialize_foo():
+            yield from _materialize("foo")
+            yield Output(1)
+
+        @op
+        def materialize_bar():
+            yield from _materialize("bar")
+            yield Output(1)
+
+        @op
+        def materialize_none():
+            yield from _materialize(None)
+            yield Output(1)
+
+        def _get_last_storage_id(storage):
+            return (
+                storage.fetch_materializations(asset_key, limit=1, ascending=False)
+                .records[0]
+                .storage_id
+            )
+
+        run_id_1, run_id_2, run_id_3 = [make_new_run_id() for i in range(3)]
+        with create_and_delete_test_runs(instance, [run_id_1, run_id_2, run_id_3]):
+            _synthesize_and_store_events(storage, lambda: materialize_foo(), run_id_1)
+
+            after_one = _get_last_storage_id(storage)
+
+            assert storage.get_updated_data_version_partitions(
+                asset_key, partitions=partitions, since_storage_id=-1
+            ) == {
+                "1",
+                "2",
+                "3",
+            }
+            assert (
+                storage.get_updated_data_version_partitions(
+                    asset_key, partitions=partitions, since_storage_id=after_one
+                )
+                == set()
+            )
+
+            # change data version
+            _synthesize_and_store_events(storage, lambda: materialize_bar(), run_id_2)
+            after_two = _get_last_storage_id(storage)
+            assert storage.get_updated_data_version_partitions(
+                asset_key, partitions=partitions, since_storage_id=after_one
+            ) == {
+                "1",
+                "2",
+                "3",
+            }
+            assert (
+                storage.get_updated_data_version_partitions(
+                    asset_key, partitions=partitions, since_storage_id=after_two
+                )
+                == set()
+            )
+
+            # materialize without data version
+            _synthesize_and_store_events(storage, lambda: materialize_bar(), run_id_3)
```

Comment:
@prha this `materialize_bar` should be `materialize_none` -- when I checked this out and made that change, the `get_updated_data_version_partitions` returned an empty set, which I think is due to a similar issue to what was described above. In this case, the latest data version storage id targets the incorrect record (the one with a None data version), meaning that when it's joined on the AssetEventTags table, there are no matching rows